internal/ops/store_op.go

```diff
@@ -1,23 +1,26 @@
 package ops
 
 import (
+	"net/http"
+
 	"github.com/dicedb/dice/internal/cmd"
 	"github.com/dicedb/dice/internal/comm"
 	"github.com/dicedb/dice/internal/eval"
 )
 
 type StoreOp struct {
-	SeqID     int16         // SeqID is the sequence id of the operation within a single request (optional, may be used for ordering)
-	RequestID int32         // RequestID identifies the request that this StoreOp belongs to
-	Cmd       *cmd.RedisCmd // Cmd is the atomic Store command (e.g., GET, SET)
-	ShardID   int           // ShardID of the shard on which the Store command will be executed
-	WorkerID  string        // WorkerID is the ID of the worker that sent this Store operation
-	Client    *comm.Client  // Client that sent this Store operation. TODO: This can potentially replace the WorkerID in the future
-	HTTPOp    bool          // HTTPOp is true if this Store operation is a HTTP operation
+	SeqID                int16               // SeqID is the sequence id of the operation within a single request (optional, may be used for ordering)
+	RequestID            uint32              // RequestID identifies the request that this StoreOp belongs to
+	Cmd                  *cmd.RedisCmd       // Cmd is the atomic Store command (e.g., GET, SET)
+	ShardID              int                 // ShardID of the shard on which the Store command will be executed
+	WorkerID             string              // WorkerID is the ID of the worker that sent this Store operation
+	Client               *comm.Client        // Client that sent this Store operation. TODO: This can potentially replace the WorkerID in the future
+	HTTPOp               bool                // HTTPOp is true if this Store operation is a HTTP operation
+	HTTPClientRespWriter http.ResponseWriter // Response channel to send back the operation response
```

Comment:
Same concern regarding io.Writer here.

What comes to my mind is an outputMode enum and an io.Writer object


Also curious if even the tcp based resp protocol can be done through io.Writer (still new to the codebase to know all the nuances)